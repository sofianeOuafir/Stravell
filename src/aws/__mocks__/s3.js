const uploadFile = () => {
  const Location = 'https://somewhere.com';
  return Promise.resolve({ Location });
}

export { uploadFile };