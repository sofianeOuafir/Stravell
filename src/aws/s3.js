import AWS from 'aws-sdk';

// Initialize the Amazon Cognito credentials provider
AWS.config.region = process.env.S3_BUCKET_REGION;
AWS.config.credentials = new AWS.CognitoIdentityCredentials({
	IdentityPoolId: process.env.S3_IDENTITY_POOL_ID,
	secretAccessKey: process.env.S3_SECRET_ACCESS_KEY,
	accessKeyId: process.env.S3_ACCESS_KEY_ID
});

const s3 = new AWS.S3({
  params: {Bucket: process.env.S3_BUCKET_NAME}
});

export const uploadFile = (file) => {
	const albumName = 'pictures'
	const { name: fileName } = file;
  var albumPhotosKey = encodeURIComponent(albumName) + '/';
	var photoKey = albumPhotosKey + fileName;
	
	return new Promise((resolve, reject) => {
		s3.upload({
			Key: photoKey,
			Body: file,
			ACL: 'public-read'
		}, function(err, data) {
			if (err) {
				reject(`There was an error when trying to upload your image`);
			} else {
				resolve(data);
			}
		});
	});
}

