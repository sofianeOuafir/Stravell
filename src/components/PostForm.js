import React from "react";
import moment from "moment";
import { EditorState, convertFromRaw, convertToRaw } from "draft-js";
import { FaCheck, FaTimes } from "react-icons/fa";
import uuid from 'uuid';
import MyEditor from "./MyEditor";
import { uploadFile } from "./../aws/s3";
import Loading from "./Loading";
import {
  formatTitle,
  formatDescription,
  getTitleError,
  getDescriptionError,
  getImageError,
  getBodyError
} from "./../lib/utils/post";

class PostForm extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      title: (props.post && props.post.title) || "",
      description: (props.post && props.post.description) || "",
      image: (props.post && props.post.image) || "",
      body:
        (props.post &&
          EditorState.createWithContent(
            convertFromRaw(JSON.parse(props.post.body))
          )) ||
        EditorState.createEmpty(),
      createdAt: (props.post && moment(props.post.createdAt)) || moment(),
      s3FolderName: (props.post && props.post.s3FolderName) || uuid(),
      updatedAt: moment(),
      titleError: "",
      descriptionError: "",
      imageError: "",
      bodyError: "",
      imageUploading: false
    };
  }

  onDescriptionChange = e => {
    const description = e.target.value;
    this.setState(() => ({
      description: description,
      descriptionError: getDescriptionError(description)
    }));
  };

  onTitleChange = e => {
    const title = e.target.value;
    this.setState(() => ({ title: title, titleError: getTitleError(title) }));
  };

  onImageChange = e => {
    const file = new File([e.target.files[0]], `main`, {type: e.target.files[0].type});
    if(!file) {
      return;
    }
    this.setState(() => ({ imageUploading: true }), () => {
      uploadFile({ file, location: `pictures/${this.state.s3FolderName}` })
        .then(({ Location }) => {
          this.setState(() => ({ image: Location, imageError: getImageError(Location), imageUploading: false }));
        })
        .catch(err => {
          alert(err);
          this.setState(() => ({ imageUploading: false }))
        });
    });
  };

  onBodyChange = editorState => {
    const body = editorState.getCurrentContent().getPlainText();
    this.setState(() => ({ body: editorState, bodyError: getBodyError(body) }));
  };

  onSubmit = e => {
    e.preventDefault();
    let errors = {
      title: getTitleError(this.state.title),
      description: getDescriptionError(this.state.description),
      image: getImageError(this.state.image),
      body: getBodyError(this.state.body.getCurrentContent().getPlainText())
    };
    if (
      errors.title === "" &&
      errors.description === "" &&
      errors.image === "" &&
      errors.body === ""
    ) {
      this.props.onSubmit({
        title: formatTitle(this.state.title),
        description: formatDescription(this.state.description),
        image: this.state.image,
        body: JSON.stringify(convertToRaw(this.state.body.getCurrentContent())),
        createdAt: this.state.createdAt.valueOf(),
        updatedAt: this.state.updatedAt.valueOf(),
        s3FolderName: this.state.s3FolderName
      });
    } else {
      this.setState(() => ({
        titleError: errors.title,
        descriptionError: errors.description,
        imageError: errors.image,
        bodyError: errors.body
      }));
    }
  }

  getValidationIcon = (error) => {
    if(error) {
      return <FaTimes className={`icon c-warm-peach`}/>;
    } else {
      return <FaCheck className={`icon c-limegreen`} />;
    }
  }

  render() {
    return (
      <form className="form" onSubmit={this.onSubmit}>
        <div className="form__input-container">
          { this.getValidationIcon(this.state.titleError) }
          <input
            placeholder="Write a title here"
            className="text-input"
            type="text"
            onChange={this.onTitleChange}
            autoFocus
            value={this.state.title}
          />
          {this.state.titleError && <p className="error">{this.state.titleError}</p>}
        </div>
        <div className="form__input-container">
          { this.getValidationIcon(this.state.descriptionError) }
          <input
            placeholder="Write a small description here"
            className="text-input"
            type="text"
            onChange={this.onDescriptionChange}
            value={this.state.description}
          />
          {this.state.descriptionError && <p className="error">{this.state.descriptionError}</p>}
        </div>
        <div className="form__input-container">
          { this.getValidationIcon(this.state.imageError) }  
          <input type="file" accept="image/*" onChange={this.onImageChange} className="text-input" />
          {this.state.imageError && <p className="error">{this.state.imageError}</p>}
          <div className="quarterwidth flex align-items--center justify-content--center mt2">
            { this.state.imageUploading ? 
              (
                <Loading size="small" />
              )
              :
              (
                this.state.image && (
                  <img src={`${this.state.image}`} alt={`${this.state.image}`} className="fullwidth" />
                )
              )
            }
          </div>

        </div>
        <div className="form__input-container">
          { this.state.bodyError && <p className="error">{this.state.bodyError}</p>}
          <MyEditor
            placeholder="Write your article here"
            editorState={this.state.body}
            onChange={this.onBodyChange}
            s3FolderName={this.state.s3FolderName}
          />
        </div>
        <div>
          <button className="button">Save Post</button>
        </div>
      </form>
    )
  }
}

export default PostForm;
