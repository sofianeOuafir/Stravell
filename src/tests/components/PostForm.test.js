import React from "react";
import { shallow } from "enzyme";
import PostForm from "./../../components/PostForm";
import {
  MIN_NUM_OF_CHARACTERS_FOR_DESCRIPTION,
  MAX_NUM_OF_CHARACTERS_FOR_DESCRIPTION,
  DESCRIPTION_TOO_SHORT_ERROR,
  DESCRIPTION_TOO_LONG_ERROR,
  TITLE_TOO_SHORT_ERROR,
  TITLE_TOO_LONG_ERROR,
  MIN_NUM_OF_CHARACTERS_FOR_TITLE,
  MAX_NUM_OF_CHARACTERS_FOR_TITLE
} from "./../../constants/constants";

import { uploadFile } from "./../../aws/s3.js";
jest.mock("./../../aws/s3.js", () => ({
  uploadFile: jest.fn()
}));

const generateTooShortString = length => {
  let word = "";
  for (var i = 0; i < length - 1; i++) {
    word += "a";
  }
  return word;
};

const generateTooLongString = length => {
  let word = "";
  for (var i = 0; i < length + 1; i++) {
    word += "a";
  }
  return word;
};

let wrapper;
beforeEach(() => {
  wrapper = shallow(<PostForm />);
});

describe("post - description", () => {
  describe("the description is changing", () => {
    const description = "A new description";
    const eventObject = { target: { value: description } };

    test("should change the state correctly", () => {
      expect(wrapper.state("description")).toEqual("");
      wrapper.find("#descriptionInput").simulate("change", eventObject);
      expect(wrapper.state("description")).toEqual(description);
    });

    test("should render correctly", () => {
      wrapper.find("#descriptionInput").simulate("change", eventObject);
      expect(wrapper).toMatchSnapshot();
    });
  });

  describe("the description is too short", () => {
    const description = generateTooShortString(
      MIN_NUM_OF_CHARACTERS_FOR_DESCRIPTION
    );
    const eventObject = { target: { value: description } };

    test("should change the state correctly", () => {
      expect(wrapper.state("description")).toEqual("");
      expect(wrapper.state("descriptionError")).toEqual("");
      wrapper.find("#descriptionInput").simulate("change", eventObject);
      expect(wrapper.state("description")).toEqual(description);
      expect(wrapper.state("descriptionError")).toEqual(
        DESCRIPTION_TOO_SHORT_ERROR
      );
    });

    test("should render correctly", () => {
      wrapper.find("#descriptionInput").simulate("change", eventObject);
      expect(wrapper).toMatchSnapshot();
    });
  });

  describe("the description is too long", () => {
    const description = generateTooLongString(
      MAX_NUM_OF_CHARACTERS_FOR_DESCRIPTION
    );
    const eventObject = { target: { value: description } };

    test("should change the state correctly", () => {
      expect(wrapper.state("description")).toEqual("");
      expect(wrapper.state("descriptionError")).toEqual("");
      wrapper.find("#descriptionInput").simulate("change", eventObject);
      expect(wrapper.state("description")).toEqual(description);
      expect(wrapper.state("descriptionError")).toEqual(
        DESCRIPTION_TOO_LONG_ERROR
      );
    });

    test("should render correctly", () => {
      wrapper.find("#descriptionInput").simulate("change", eventObject);
      expect(wrapper).toMatchSnapshot();
    });
  });
});

describe("post - title", () => {
  describe("the title is changing", () => {
    const title = "A new title";
    const eventObject = { target: { value: title } };

    test("should change the state correctly", () => {
      expect(wrapper.state("title")).toEqual("");
      wrapper.find("#titleInput").simulate("change", eventObject);
      expect(wrapper.state("title")).toEqual(title);
    });

    test("should render correctly", () => {
      wrapper.find("#titleInput").simulate("change", eventObject);
      expect(wrapper).toMatchSnapshot();
    });
  });

  describe("the title is too short", () => {
    const title = generateTooShortString(MIN_NUM_OF_CHARACTERS_FOR_TITLE);
    const eventObject = { target: { value: title } };

    test("should change the state correctly", () => {
      expect(wrapper.state("title")).toEqual("");
      expect(wrapper.state("titleError")).toEqual("");
      wrapper.find("#titleInput").simulate("change", eventObject);
      expect(wrapper.state("title")).toEqual(title);
      expect(wrapper.state("titleError")).toEqual(TITLE_TOO_SHORT_ERROR);
    });

    test("should render correctly", () => {
      wrapper.find("#titleInput").simulate("change", eventObject);
      expect(wrapper).toMatchSnapshot();
    });
  });

  describe("the title is too long", () => {
    const title = generateTooLongString(MAX_NUM_OF_CHARACTERS_FOR_TITLE);
    const eventObject = { target: { value: title } };

    test("should change the state correctly", () => {
      expect(wrapper.state("title")).toEqual("");
      expect(wrapper.state("titleError")).toEqual("");
      wrapper.find("#titleInput").simulate("change", eventObject);
      expect(wrapper.state("title")).toEqual(title);
      expect(wrapper.state("titleError")).toEqual(TITLE_TOO_LONG_ERROR);
    });

    test("should render correctly", () => {
      wrapper.find("#titleInput").simulate("change", eventObject);
      expect(wrapper).toMatchSnapshot();
    });
  });
});

describe("post - image", () => {
  const file = { type: ".png" };
  const eventObject = { target: { files: [file] } };

  describe("the image get changed", () => {
    beforeEach(() => {
      uploadFile.mockImplementation(() => {
        const Location = 'https://somewhere.com';
        return Promise.resolve({ Location });
      });
    });

    test("should change the state correctly", done => {
      wrapper.find("#imageInput").simulate("change", eventObject);
      return Promise.resolve().then(() => {
        expect(wrapper.state("image")).toEqual("https://somewhere.com");
        done();
      });
    });

    test("should render correctly", done => {
      wrapper.find("#imageInput").simulate("change", eventObject);
      return Promise.resolve().then(() => {
        wrapper.update();
        expect(wrapper).toMatchSnapshot();
        done();
      });
    });
  });

  describe('An error occured when the image get uploaded', () => {
    beforeEach(() => {
      uploadFile.mockImplementation(() => {
        return Promise.reject('an error occurred');
      });
    });

    test("should change the state correctly", (done) => {
      wrapper.find("#imageInput").simulate("change", eventObject);
      return Promise.resolve().then(() => {
        wrapper.update();
        expect(wrapper.state("imageError")).toEqual('an error occurred');
        done();
      })
    });

    test('should render correctly', (done) => {
      wrapper.find("#imageInput").simulate("change", eventObject);
      return Promise.resolve().then(() => {
        expect(wrapper).toMatchSnapshot();
        done();
      });
    });
  });
});
