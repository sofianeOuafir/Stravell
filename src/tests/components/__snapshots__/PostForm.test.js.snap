// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`post - description the description is changing should render correctly 1`] = `
<form
  className="form mb3 pb3"
  onSubmit={[Function]}
>
  <div
    className="form__input-container"
  >
    <FaCheck
      className="icon c-limegreen"
    />
    <input
      autoFocus={true}
      className="text-input"
      id="titleInput"
      onChange={[Function]}
      placeholder="Write a title here"
      type="text"
      value=""
    />
  </div>
  <div
    className="form__input-container"
  >
    <FaCheck
      className="icon c-limegreen"
    />
    <input
      className="text-input"
      id="descriptionInput"
      onChange={[Function]}
      placeholder="Write a small description here"
      type="text"
      value="A new description"
    />
  </div>
  <div
    className="form__input-container"
  >
    <FaCheck
      className="icon c-limegreen"
    />
    <input
      accept="image/*"
      className="text-input"
      id="imageInput"
      onChange={[Function]}
      type="file"
    />
    <div
      className="quarterwidth flex align-items--center justify-content--center mt2"
    />
  </div>
  <div>
    <p>
      What do you want to do today?
    </p>
    <Checkbox
      checked={true}
      handleChange={[Function]}
      label="I want to write the content of the post."
    />
    <Checkbox
      checked={false}
      handleChange={[Function]}
      label="I want to provide an URL for this post (users will be redirected to the provided URL). "
    />
  </div>
  <div
    className="form__input-container"
  >
    <MyEditor
      editorState={Object {}}
      onChange={[Function]}
      placeholder="Write your article here"
      s3FolderName="abc123"
    />
  </div>
  <div>
    <button
      className="button"
    >
      Save Post
    </button>
  </div>
</form>
`;

exports[`post - description the description is too long should render correctly 1`] = `
<form
  className="form mb3 pb3"
  onSubmit={[Function]}
>
  <div
    className="form__input-container"
  >
    <FaCheck
      className="icon c-limegreen"
    />
    <input
      autoFocus={true}
      className="text-input"
      id="titleInput"
      onChange={[Function]}
      placeholder="Write a title here"
      type="text"
      value=""
    />
  </div>
  <div
    className="form__input-container"
  >
    <FaTimes
      className="icon c-warm-peach"
    />
    <input
      className="text-input"
      id="descriptionInput"
      onChange={[Function]}
      placeholder="Write a small description here"
      type="text"
      value="aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa"
    />
    <p
      className="error"
    >
      Description should be maximum 180 characters long
    </p>
  </div>
  <div
    className="form__input-container"
  >
    <FaCheck
      className="icon c-limegreen"
    />
    <input
      accept="image/*"
      className="text-input"
      id="imageInput"
      onChange={[Function]}
      type="file"
    />
    <div
      className="quarterwidth flex align-items--center justify-content--center mt2"
    />
  </div>
  <div>
    <p>
      What do you want to do today?
    </p>
    <Checkbox
      checked={true}
      handleChange={[Function]}
      label="I want to write the content of the post."
    />
    <Checkbox
      checked={false}
      handleChange={[Function]}
      label="I want to provide an URL for this post (users will be redirected to the provided URL). "
    />
  </div>
  <div
    className="form__input-container"
  >
    <MyEditor
      editorState={Object {}}
      onChange={[Function]}
      placeholder="Write your article here"
      s3FolderName="abc123"
    />
  </div>
  <div>
    <button
      className="button"
    >
      Save Post
    </button>
  </div>
</form>
`;

exports[`post - description the description is too short should render correctly 1`] = `
<form
  className="form mb3 pb3"
  onSubmit={[Function]}
>
  <div
    className="form__input-container"
  >
    <FaCheck
      className="icon c-limegreen"
    />
    <input
      autoFocus={true}
      className="text-input"
      id="titleInput"
      onChange={[Function]}
      placeholder="Write a title here"
      type="text"
      value=""
    />
  </div>
  <div
    className="form__input-container"
  >
    <FaTimes
      className="icon c-warm-peach"
    />
    <input
      className="text-input"
      id="descriptionInput"
      onChange={[Function]}
      placeholder="Write a small description here"
      type="text"
      value="aaaa"
    />
    <p
      className="error"
    >
      Description should be minimum 5 characters long
    </p>
  </div>
  <div
    className="form__input-container"
  >
    <FaCheck
      className="icon c-limegreen"
    />
    <input
      accept="image/*"
      className="text-input"
      id="imageInput"
      onChange={[Function]}
      type="file"
    />
    <div
      className="quarterwidth flex align-items--center justify-content--center mt2"
    />
  </div>
  <div>
    <p>
      What do you want to do today?
    </p>
    <Checkbox
      checked={true}
      handleChange={[Function]}
      label="I want to write the content of the post."
    />
    <Checkbox
      checked={false}
      handleChange={[Function]}
      label="I want to provide an URL for this post (users will be redirected to the provided URL). "
    />
  </div>
  <div
    className="form__input-container"
  >
    <MyEditor
      editorState={Object {}}
      onChange={[Function]}
      placeholder="Write your article here"
      s3FolderName="abc123"
    />
  </div>
  <div>
    <button
      className="button"
    >
      Save Post
    </button>
  </div>
</form>
`;

exports[`post - image An error occured when the image got uploaded should render correctly 1`] = `
<form
  className="form mb3 pb3"
  onSubmit={[Function]}
>
  <div
    className="form__input-container"
  >
    <FaCheck
      className="icon c-limegreen"
    />
    <input
      autoFocus={true}
      className="text-input"
      id="titleInput"
      onChange={[Function]}
      placeholder="Write a title here"
      type="text"
      value=""
    />
  </div>
  <div
    className="form__input-container"
  >
    <FaCheck
      className="icon c-limegreen"
    />
    <input
      className="text-input"
      id="descriptionInput"
      onChange={[Function]}
      placeholder="Write a small description here"
      type="text"
      value=""
    />
  </div>
  <div
    className="form__input-container"
  >
    <FaTimes
      className="icon c-warm-peach"
    />
    <input
      accept="image/*"
      className="text-input"
      id="imageInput"
      onChange={[Function]}
      type="file"
    />
    <p
      className="error"
    >
      an error occurred
    </p>
    <div
      className="quarterwidth flex align-items--center justify-content--center mt2"
    />
  </div>
  <div>
    <p>
      What do you want to do today?
    </p>
    <Checkbox
      checked={true}
      handleChange={[Function]}
      label="I want to write the content of the post."
    />
    <Checkbox
      checked={false}
      handleChange={[Function]}
      label="I want to provide an URL for this post (users will be redirected to the provided URL). "
    />
  </div>
  <div
    className="form__input-container"
  >
    <MyEditor
      editorState={Object {}}
      onChange={[Function]}
      placeholder="Write your article here"
      s3FolderName="abc123"
    />
  </div>
  <div>
    <button
      className="button"
    >
      Save Post
    </button>
  </div>
</form>
`;

exports[`post - image the format of the file is not an accepted format should render correctly 1`] = `
<form
  className="form mb3 pb3"
  onSubmit={[Function]}
>
  <div
    className="form__input-container"
  >
    <FaCheck
      className="icon c-limegreen"
    />
    <input
      autoFocus={true}
      className="text-input"
      id="titleInput"
      onChange={[Function]}
      placeholder="Write a title here"
      type="text"
      value=""
    />
  </div>
  <div
    className="form__input-container"
  >
    <FaCheck
      className="icon c-limegreen"
    />
    <input
      className="text-input"
      id="descriptionInput"
      onChange={[Function]}
      placeholder="Write a small description here"
      type="text"
      value=""
    />
  </div>
  <div
    className="form__input-container"
  >
    <FaTimes
      className="icon c-warm-peach"
    />
    <input
      accept="image/*"
      className="text-input"
      id="imageInput"
      onChange={[Function]}
      type="file"
    />
    <p
      className="error"
    >
      The format of the uploaded image is not accepted.
    </p>
    <div
      className="quarterwidth flex align-items--center justify-content--center mt2"
    />
  </div>
  <div>
    <p>
      What do you want to do today?
    </p>
    <Checkbox
      checked={true}
      handleChange={[Function]}
      label="I want to write the content of the post."
    />
    <Checkbox
      checked={false}
      handleChange={[Function]}
      label="I want to provide an URL for this post (users will be redirected to the provided URL). "
    />
  </div>
  <div
    className="form__input-container"
  >
    <MyEditor
      editorState={Object {}}
      onChange={[Function]}
      placeholder="Write your article here"
      s3FolderName="abc123"
    />
  </div>
  <div>
    <button
      className="button"
    >
      Save Post
    </button>
  </div>
</form>
`;

exports[`post - image the image got changed and uploaded successfully should render correctly 1`] = `
<form
  className="form mb3 pb3"
  onSubmit={[Function]}
>
  <div
    className="form__input-container"
  >
    <FaCheck
      className="icon c-limegreen"
    />
    <input
      autoFocus={true}
      className="text-input"
      id="titleInput"
      onChange={[Function]}
      placeholder="Write a title here"
      type="text"
      value=""
    />
  </div>
  <div
    className="form__input-container"
  >
    <FaCheck
      className="icon c-limegreen"
    />
    <input
      className="text-input"
      id="descriptionInput"
      onChange={[Function]}
      placeholder="Write a small description here"
      type="text"
      value=""
    />
  </div>
  <div
    className="form__input-container"
  >
    <FaCheck
      className="icon c-limegreen"
    />
    <input
      accept="image/*"
      className="text-input"
      id="imageInput"
      onChange={[Function]}
      type="file"
    />
    <div
      className="quarterwidth flex align-items--center justify-content--center mt2"
    >
      <img
        alt="https://somewhere.com"
        className="fullwidth"
        src="https://somewhere.com"
      />
    </div>
  </div>
  <div>
    <p>
      What do you want to do today?
    </p>
    <Checkbox
      checked={true}
      handleChange={[Function]}
      label="I want to write the content of the post."
    />
    <Checkbox
      checked={false}
      handleChange={[Function]}
      label="I want to provide an URL for this post (users will be redirected to the provided URL). "
    />
  </div>
  <div
    className="form__input-container"
  >
    <MyEditor
      editorState={Object {}}
      onChange={[Function]}
      placeholder="Write your article here"
      s3FolderName="abc123"
    />
  </div>
  <div>
    <button
      className="button"
    >
      Save Post
    </button>
  </div>
</form>
`;

exports[`post - title the title is changing should render correctly 1`] = `
<form
  className="form mb3 pb3"
  onSubmit={[Function]}
>
  <div
    className="form__input-container"
  >
    <FaCheck
      className="icon c-limegreen"
    />
    <input
      autoFocus={true}
      className="text-input"
      id="titleInput"
      onChange={[Function]}
      placeholder="Write a title here"
      type="text"
      value="A new title"
    />
  </div>
  <div
    className="form__input-container"
  >
    <FaCheck
      className="icon c-limegreen"
    />
    <input
      className="text-input"
      id="descriptionInput"
      onChange={[Function]}
      placeholder="Write a small description here"
      type="text"
      value=""
    />
  </div>
  <div
    className="form__input-container"
  >
    <FaCheck
      className="icon c-limegreen"
    />
    <input
      accept="image/*"
      className="text-input"
      id="imageInput"
      onChange={[Function]}
      type="file"
    />
    <div
      className="quarterwidth flex align-items--center justify-content--center mt2"
    />
  </div>
  <div>
    <p>
      What do you want to do today?
    </p>
    <Checkbox
      checked={true}
      handleChange={[Function]}
      label="I want to write the content of the post."
    />
    <Checkbox
      checked={false}
      handleChange={[Function]}
      label="I want to provide an URL for this post (users will be redirected to the provided URL). "
    />
  </div>
  <div
    className="form__input-container"
  >
    <MyEditor
      editorState={Object {}}
      onChange={[Function]}
      placeholder="Write your article here"
      s3FolderName="abc123"
    />
  </div>
  <div>
    <button
      className="button"
    >
      Save Post
    </button>
  </div>
</form>
`;

exports[`post - title the title is too long should render correctly 1`] = `
<form
  className="form mb3 pb3"
  onSubmit={[Function]}
>
  <div
    className="form__input-container"
  >
    <FaTimes
      className="icon c-warm-peach"
    />
    <input
      autoFocus={true}
      className="text-input"
      id="titleInput"
      onChange={[Function]}
      placeholder="Write a title here"
      type="text"
      value="aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa"
    />
    <p
      className="error"
    >
      Title should be maximum 80 characters long
    </p>
  </div>
  <div
    className="form__input-container"
  >
    <FaCheck
      className="icon c-limegreen"
    />
    <input
      className="text-input"
      id="descriptionInput"
      onChange={[Function]}
      placeholder="Write a small description here"
      type="text"
      value=""
    />
  </div>
  <div
    className="form__input-container"
  >
    <FaCheck
      className="icon c-limegreen"
    />
    <input
      accept="image/*"
      className="text-input"
      id="imageInput"
      onChange={[Function]}
      type="file"
    />
    <div
      className="quarterwidth flex align-items--center justify-content--center mt2"
    />
  </div>
  <div>
    <p>
      What do you want to do today?
    </p>
    <Checkbox
      checked={true}
      handleChange={[Function]}
      label="I want to write the content of the post."
    />
    <Checkbox
      checked={false}
      handleChange={[Function]}
      label="I want to provide an URL for this post (users will be redirected to the provided URL). "
    />
  </div>
  <div
    className="form__input-container"
  >
    <MyEditor
      editorState={Object {}}
      onChange={[Function]}
      placeholder="Write your article here"
      s3FolderName="abc123"
    />
  </div>
  <div>
    <button
      className="button"
    >
      Save Post
    </button>
  </div>
</form>
`;

exports[`post - title the title is too short should render correctly 1`] = `
<form
  className="form mb3 pb3"
  onSubmit={[Function]}
>
  <div
    className="form__input-container"
  >
    <FaTimes
      className="icon c-warm-peach"
    />
    <input
      autoFocus={true}
      className="text-input"
      id="titleInput"
      onChange={[Function]}
      placeholder="Write a title here"
      type="text"
      value="aaaa"
    />
    <p
      className="error"
    >
      Title should be minimum 5 characters long
    </p>
  </div>
  <div
    className="form__input-container"
  >
    <FaCheck
      className="icon c-limegreen"
    />
    <input
      className="text-input"
      id="descriptionInput"
      onChange={[Function]}
      placeholder="Write a small description here"
      type="text"
      value=""
    />
  </div>
  <div
    className="form__input-container"
  >
    <FaCheck
      className="icon c-limegreen"
    />
    <input
      accept="image/*"
      className="text-input"
      id="imageInput"
      onChange={[Function]}
      type="file"
    />
    <div
      className="quarterwidth flex align-items--center justify-content--center mt2"
    />
  </div>
  <div>
    <p>
      What do you want to do today?
    </p>
    <Checkbox
      checked={true}
      handleChange={[Function]}
      label="I want to write the content of the post."
    />
    <Checkbox
      checked={false}
      handleChange={[Function]}
      label="I want to provide an URL for this post (users will be redirected to the provided URL). "
    />
  </div>
  <div
    className="form__input-container"
  >
    <MyEditor
      editorState={Object {}}
      onChange={[Function]}
      placeholder="Write your article here"
      s3FolderName="abc123"
    />
  </div>
  <div>
    <button
      className="button"
    >
      Save Post
    </button>
  </div>
</form>
`;
